test/analysis-tests.js

```diff
@@ -31,16 +31,19 @@ const languages = {
     analysis: {id: 'analysis'}
 };
 
-const compilerProps = new properties.CompilerProps(languages, properties.fakeProps({}));
-
 describe('LLVM-mca tool definition', () => {
-    const ce = new CompilationEnvironment(compilerProps);
-    const info = {
-        exe: null,
-        remote: true,
-        lang: languages.analysis.id
-    };
-    const a = new LLVMmcaTool(info, ce);
+    let ce, a;
```

Comment:
Neat! Thanks. I'm still learning how best to do all this stuff in JS.

The "`before`" stuff is per-scope of `describe`, right? These aren't going to leak out between test JS files somehow?

Reply:
`before` will be executed once before all tests in the containing `describe` block, correct.

However, all tests are executed in a single node process serially, so there is nothing to stop global state leakage between tests.

I actually think we already have some test setup leakage already, as if you do something like `find test/ -name '*.js' -exec mocha {} \;` to execute each test file on its own some tests are failing. I didn't investigate this further, but it's something to think about in the future.